streams/src/test/java/org/apache/kafka/streams/state/internals/InMemoryKeyValueLoggedStoreTest.java

```diff
@@ -17,44 +17,197 @@
+    }
+
+    @Test
+    public void shouldReturnCorrectApproximateNumberOfEntries() {
+        innerKeyValueStore.put(1, "one");
+        innerKeyValueStore.put(2, "two");
+        EasyMock.expect(innerKeyValueStore.approximateNumEntries()).andReturn(2L);
+        EasyMock.replay(innerKeyValueStore);
+
+        inMemoryKeyValueLoggedStore.init(context, innerKeyValueStore);
+        inMemoryKeyValueLoggedStore.put(1, "one");
+        inMemoryKeyValueLoggedStore.put(2, "two");
+
+        assertThat(inMemoryKeyValueLoggedStore.approximateNumEntries(), is(2L));
+        EasyMock.verify(innerKeyValueStore);
     }
+
+    @Test
+    public void shouldReturnAll() {
+        final KeyValueIteratorStub<Integer, String>
+            iteratorStub =
+            new KeyValueIteratorStub<>(Arrays.asList(KeyValue.pair(1, "one"), KeyValue.pair(2, "two")).iterator());
+        innerKeyValueStore.put(1, "one");
+        innerKeyValueStore.put(2, "two");
+        EasyMock.expect(innerKeyValueStore.all()).andReturn(iteratorStub);
+        EasyMock.replay(innerKeyValueStore);
+
+        inMemoryKeyValueLoggedStore.init(context, innerKeyValueStore);
+        inMemoryKeyValueLoggedStore.put(1, "one");
+        inMemoryKeyValueLoggedStore.put(2, "two");
+
+        final List<KeyValue<Integer, String>> allReturned = new ArrayList<>();
+        final List<KeyValue<Integer, String>> expectedReturned = Arrays.asList(KeyValue.pair(1, "one"), KeyValue.pair(2, "two"));
+        final Iterator<KeyValue<Integer, String>> iterator = inMemoryKeyValueLoggedStore.all();
+
+        while (iterator.hasNext()) {
+            allReturned.add(iterator.next());
+        }
+
+        assertThat(allReturned, equalTo(expectedReturned));
+        EasyMock.verify(innerKeyValueStore);
+    }
+
+    @Test
+    public void shouldPutAllInLoggedStore() {
+        final KeyValueIteratorStub<Integer, String>
+            iteratorStub =
+            new KeyValueIteratorStub<>(Arrays.asList(KeyValue.pair(1, "one"), KeyValue.pair(2, "two")).iterator());
+        innerKeyValueStore.putAll(Arrays.asList(KeyValue.pair(1, "one"), KeyValue.pair(2, "two")));
+        EasyMock.expect(innerKeyValueStore.all()).andReturn(iteratorStub);
+        EasyMock.replay(innerKeyValueStore);
+
+        inMemoryKeyValueLoggedStore.init(context, innerKeyValueStore);
+
+        final List<KeyValue<Integer, String>> allReturned = new ArrayList<>();
+        final List<KeyValue<Integer, String>> initialPutAll = Arrays.asList(KeyValue.pair(1, "one"), KeyValue.pair(2, "two"));
+
+        inMemoryKeyValueLoggedStore.putAll(initialPutAll);
+
+        final Iterator<KeyValue<Integer, String>> iterator = inMemoryKeyValueLoggedStore.all();
+
+        while (iterator.hasNext()) {
+            allReturned.add(iterator.next());
+        }
+        assertThat(allReturned, equalTo(initialPutAll));
+        EasyMock.verify(innerKeyValueStore);
+    }
+
+    @Test
+    public void shouldPutIfAbsentValidUpdate() {
```

Comment:
As mentioned below, can this test and the other ones in RocksDBStore etc be merged into a single test for the parent class?

Reply:
test already exists in `AbstractKeyValueStore`  deleting.